src/pstl_offload.cpp

```diff
@@ -29,12 +29,49 @@
 namespace __pstl_offload
 {
 
-static auto
-__get_original_free()
+using __free_func_type = void (*)(void*);
+
+// list of objects for delayed releasing
+struct __delayed_free_list {
+    __delayed_free_list* _M_next;
+    void*                  _M_to_free;
+
+    __delayed_free_list(__delayed_free_list* __next, void* __to_free)
+        : _M_next(__next), _M_to_free(__to_free) {}
```

Comment:
Since `__delayed_free_list` is a trivial structure, may be it makes sense to remove the constructor (and corresponding placement new calls) and either assign fields or use aggregate initialization (similar to how `__block_header` is operated)

Reply:
Done.